### PR TITLE
Add ocid.spec to build rpms

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Dan Walsh dwalsh@redhat.com

### CODE REVIEW

**Comment by mrunalp** [comment]
@lsm5 can you look this over?

**Comment by lsm5** [comment]
@rhatdan @mrunalp I feel we should skip the specfile from upstream, since it belongs in a distro's dist-git. Else it's like: specfile in dist-git fetches upstream tarball which then again contains the specfile.

We could use this specfile separately for package review process though.

**Comment by lsm5** [comment]
@rhatdan Is the `make rpm` step something that's commonly used and something which can't be replaced by just an `rpmbuild -ba ocid.spec`. 

Also, with `make dist`, the fetched tarball will also fetch a tarball with an older commit (thus older specfile) than what you have on your local copy, so that's again why I don't quite support spec files in upstream.

File: ocid.spec [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+%global import_path     %{provider_prefix}
+%global commit          4fe9f0951eac53b180aa197257ab57af9ac0b6f8
+%global shortcommit     %(c=%{commit}; echo ${c:0:7})
+
+Name:           ocid
+Version:        0.1.1
+Release:        1.git%{shortcommit}%{?dist}
+Summary:        OCI systemd hook for docker
+Group:          Applications/Text
+License:        GPLv3+
```
Comment (cyphar) [inline ocid.spec]: I don't think this license or version is correct.

Comment (cyphar) [inline ocid.spec]: I don't think this license or version is correct.

File: ocid.spec [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+%make_install
+
+#define license tag if not already defined
+%{!?_licensedir:%global license %doc}
+%files
+%{_bindir}/ocid
+%{_bindir}/ocic
+%{_mandir}/man5/ocid.conf.5*
+%{_mandir}/man8/ocid.8*
+%{_sysconfdir}/ocid.conf
```
Comment (cyphar) [inline ocid.spec]: This should be `%config(noreplace)`.

File: ocid.spec [hunk 3]
```diff
@@ -0,0 +1,62 @@
+%define debug_package %{nil}
```
Comment (cyphar) [inline ocid.spec]: This entire file should be in `contrib/rpm/redhat/...` or something like that. Not in the main directory. This actually also applies for `ocid.service` which should be in `contrib/systemd`.

File: ocid.spec (hunk 1 above)
Comment (cyphar) [inline ocid.spec]: Actually this entire section is wrong (group, summary, release, etc).

File: Makefile [hunk 4]
```diff
@@ -45,7 +45,7 @@ ocic: ${OCID_LINK}
 	go build -o ocic ./cmd/client/
 
 ocid.conf: ocid
-	 ./ocid --config="" config --default > ocid.conf 
+	 ./ocid --config="" config --default > ocid.conf
```
Comment (cyphar) [inline Makefile]: `--config=""` is not necessary.

File: Makefile [hunk 5]
```diff
@@ -87,17 +87,17 @@ docs/%.5: docs/%.5.md
 
 docs: $(MANPAGES_MD:%.md=%)
 
-install: ocid.conf
+install:
```
Comment (cyphar) [inline Makefile]: Why drop the requirement? In fact, we should add `binaries` and `docs` requirements as well.

Comment (runcom) [inline Makefile]: to install? I don't like that, you usually do "make && sudo make install" - re-compiling on make install?

Comment (cyphar) [inline Makefile]: I don't personally do it that way, but do you really intend to do `make binaries docs ocid.conf && sudo make install` each time? The point is that doing it like this ensures that you don't install out-of-date binaries (though we need to fix up the dependencies of the binaries to include their sources).

**Comment by cyphar** [comment]
I've been working along similar lines for openSUSE. I've tried to make it as diverse as possible, so that we can in principle use the same config for both RedHat and SUSE distributions. https://build.opensuse.org/package/show/home:cyphar/cri-o

I still need to make some of the options configurable, but we'll see how that goes.

**Comment by rhatdan** [comment]
Ok I fixed up the Makefile to not build everything everytime.  And updated the spec file.  I have no problem with keeping the spec file separate, as I agree with @lsm5 that it is a Pain in the butt. 
Although it makes it easier to review  in git.  I like what @cyphar has for a spec file, although I would prefer to keep CRI-O out of names as long as possible until we get a better name.

**Comment by rhatdan** [comment]
@runcom see if the latest Makefile patches work for you.
make; make install 
should only build once.

**Comment by cyphar** [comment]
I have no problem putting the spec file in `contrib/` (see #143), and I think that could work as a baseline for distros (if we can't make it generic enough to work for both RH and SUSE -- if we can make it generic enough we can have `contrib/rpm/ocid.spec).

The naming is a valid point. I don't like the name cri-o either and we should definitely fix that soon. I'm just using it as the name for now because that's what the repo is called. :P

**Comment by rhatdan** [comment]
I removed the systemctl calls from the Makefile.

**Comment by rhatdan** [comment]
@lsm5 PTAL

**Comment by lsm5** [comment]
@rhatdan i'll use this spec file with some additions for fedora review. Thanks

**Comment by runcom** [comment]
@lsm5 @rhatdan any update here?

**Comment by rhatdan** [comment]
@runcom lets merge and then we can update it with whatever the Fedora Packaging Review forces us to change.  But I would like to get this going so we could ship it in Fedora 26.

**Comment by runcom** [comment]
@rhatdan alright, @cyphar could you approve the review and merge this?

**Comment by rhatdan** [comment]
@cyphar What do you want me to do with this?

**Comment by cyphar** [comment]
Oh, I forgot to merge it. Yeah, LGTM we can fix this later.

**Pull request merged**